src/CurrentIdentity/Storage/SessionCurrentIdentityStorage.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\User\CurrentIdentity\Storage;
+
+use Yiisoft\Session\SessionInterface;
+
+/**
+ * Stores current identity ID in a session.
+ */
+final class SessionCurrentIdentityStorage implements CurrentIdentityStorageInterface
+{
+    private const SESSION_AUTH_ID = '__auth_id';
+    private const SESSION_AUTH_EXPIRE = '__auth_expire';
+    private const SESSION_AUTH_ABSOLUTE_EXPIRE = '__auth_absolute_expire';
+
+    /**
+     * @var int|null the number of seconds in which the user will be logged out automatically in case of
+     * remaining inactive. If this property is not set, the user will be logged out after
+     * the current session expires.
+     */
+    private ?int $authTimeout = null;
+
+    /**
+     * @var int|null the number of seconds in which the user will be logged out automatically
+     * regardless of activity.
+     */
+    private ?int $absoluteAuthTimeout = null;
+
+    private SessionInterface $session;
+
+    public function __construct(SessionInterface $session)
+    {
+        $this->session = $session;
+    }
+
+    public function setAuthTimeout(int $timeout = null): self
```

Comment:
It's not part of the interface. Should it be supported by other storages and be part of the interface or not?

Reply:
This method not need in other places. I think use this method only for concrete implementation.